services/navigation/builtin/builtin.go

```diff
@@ -206,12 +219,13 @@ func NewBuiltIn(
 
 type builtIn struct {
 	resource.Named
-	actionMu  sync.RWMutex
-	mu        sync.RWMutex
-	store     navigation.NavStore
-	storeType string
-	mode      navigation.Mode
-	mapType   navigation.MapType
+	activeExecutionWaypoint atomic.Value
```

Comment:
`activeExecutionWaypoint` is added as an atomic value so it can be updated across threads without needing to take a lock on `mu`, which was useful in preventing unnecessary race conditions & deadlocks.